### PR TITLE
Update to Terraform v1.0.0, use Dependabot

### PR DESCRIPTION
**Pull request by GaryGSC**
Closes: #226
Refs: #228

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -22,6 +22,30 @@ updates:
...
+#    directory: '/terraform-iac/dev/app'
+#    schedule:
+#      interval: daily
+#    target-branch: dev
+
+#  - package-ecosystem: terraform
+#    directory: '/terraform-iac/dev/setup'
+#    schedule:
+#      interval: daily
+#    target-branch: dev
```
Comment (GaryGSC) [inline .github/dependabot.yml:35]: I don't love the idea of getting 8 PRs whenever we need to update the AWS provider, but I can't think of a way around that until https://github.com/dependabot/dependabot-core/issues/2178 is added.

**Comment by GaryGSC** [comment]
Are we thinking that the Dependabot stuff is blocked on #212?

**Comment by GaryGSC** [comment]
It looks like the main difference to the TF plan is the way we now mark more things as sensitive.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -22,6 +22,30 @@ updates:
...
+
+  - package-ecosystem: terraform
+    directory: '/terraform-iac/modules/app'
+    schedule:
+      interval: daily
```
Comment (yoshutch) [inline .github/dependabot.yml:40]: (nit) is there a reason this is daily? I've been doing weekly just so that I don't get overwhelmed with updates.

**Review by yoshutch — APPROVED** [review]

**Review by stuft2 — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (GaryGSC) [reply .github/dependabot.yml:40]: All(?) of the Terraform modules we're using are ones we've written. They don't update very often. 🤷‍♂️

**Pull request merged**